Source/SessionManager.swift

```diff
@@ -874,12 +874,16 @@ open class SessionManager {
                     return
                 }
 
-                let retrySucceeded = strongSelf.retry(request)
+                DispatchQueue.utility.after(timeDelay) {
```

Comment:
Great question. All this logic does is dispatch after a given interval onto the utility queue. We're not creating a new queue. The utility queue is concurrent and is merely a thread pool that will execute once it has resources. Generally this will happen immediately, unless you have multiple higher priority operations running.

None of this retry logic executes on the calling thread. If you create a request that fails to be adapted (for the sake of argument let's say your OAuth token is expired), then the retry logic hops off the calling thread in the `allowRetrier(_:toRetryRequest:withError:)` API. Then all the retry logic is executed on the utility queue. Once the completion is called (we don't know what queue we'll be called on), then we dispatch to utility once more to allow the delay to pass before retrying. The retry logic simply re-adapts the request and hands the task over to the session to execute. The new task will then be executed on the URLSession's internal queue and all callbacks will again execute on the SessionDelegate queue.

I think putting together a queue diagram is a great idea and is something we should add to the Trello backlog.